# Median, Mean, Mode, Percentile

We will use **Median, Mean, Mode** to fill the missing values.**Percentile** used to understand and interpret data.

# Mean

In [1]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/laxmankusuma/Data-Science-series/main/Mathematics%20and%20Statistics%20for%20Data%20Science/mean",sep=',')
df

name  salary
0     laxman  1000.0
1   santhosh  2000.0
2      ashok  3000.0
3     rajesk     NaN
4  prashanth  2000.0
5      kumar  1000.0

In [2]:
mean_value = df['salary'].mean()
mean_value

1800.0

In [3]:
df['salary'].fillna(value=mean_value, inplace=True)

In [4]:
df

name  salary
0     laxman  1000.0
1   santhosh  2000.0
2      ashok  3000.0
3     rajesk  1800.0
4  prashanth  2000.0
5      kumar  1000.0

salary of rajesh filled with mean value.

# Median use

In [5]:
df = pd.read_csv("https://raw.githubusercontent.com/laxmankusuma/Data-Science-series/main/Mathematics%20and%20Statistics%20for%20Data%20Science/median",sep=',')
pd.set_option('display.float_format', lambda x: '%0.4f' % x)
df

name           salary
0     laxman        1000.0000
1   santhosh        2000.0000
2      ashok        3000.0000
3     rajesk              NaN
4  prashanth        2000.0000
5      kumar 10000000000.0000

In [6]:
df['salary'].mean()

2000001600.0

In the above case we can not impute with mean, we use median.

The median is the middle number in a sorted, ascending or descending, list of numbers

if middle numbers are 2 then average it.

In [7]:
median_value = df['salary'].median()

In [8]:
df['salary'].fillna(value=median_value, inplace=True)
df

name           salary
0     laxman        1000.0000
1   santhosh        2000.0000
2      ashok        3000.0000
3     rajesk        2000.0000
4  prashanth        2000.0000
5      kumar 10000000000.0000

Rajesh salary updated with median

# Mode

The mode is the value that appears most often in a set of data values.

In [9]:
df = pd.read_csv("https://raw.githubusercontent.com/laxmankusuma/Data-Science-series/main/Mathematics%20and%20Statistics%20for%20Data%20Science/mode",sep=',')
df

name  color
0     laxman  brown
1     rajesh  brown
2      kumar  brown
3        bob  white
4  prashanth  brown
5   santhosh    NaN
6      ashok  brown

In [10]:
mode_value = df['color'].mode()[0]
mode_value

'brown'

In [11]:
df['color'].fillna(value=mode_value, inplace=True)
df

name  color
0     laxman  brown
1     rajesh  brown
2      kumar  brown
3        bob  white
4  prashanth  brown
5   santhosh  brown
6      ashok  brown

santhosh color updated with brown color.

# Percentile

A percentile is based on the rank of numbers.

we can remove outliers using percentile

In [12]:
df = pd.read_csv("https://raw.githubusercontent.com/laxmankusuma/Data-Science-series/main/Mathematics%20and%20Statistics%20for%20Data%20Science/Percentile",sep=',')
df

key  value
0     1   1000
1     2   2000
2     3   3000
3     4    500
4     5   4000
5     6   1200
6     7   2900
7     8   1100
8     9   5000
9    10    800
10   11    900
11   12   6000

sort all the values

500,800, 900, 1000, 1100,1200,2000,2900,3000,4000, 5000,6000



borders of the percentile , we call as quantile

In [13]:
df['value'].quantile(0.25)

975.0

In [14]:
df['value'].quantile(0.50) 

1600.0

In [15]:
df['value'].quantile(0.75) 

3250.0

In [16]:
df['value'].quantile(1) # big value

6000.0

In [17]:
df['value'].quantile(0) #  small value

500.0

In [18]:
df.describe()

key     value
count 12.0000   12.0000
mean   6.5000 2366.6667
std    3.6056 1822.7518
min    1.0000  500.0000
25%    3.7500  975.0000
50%    6.5000 1600.0000
75%    9.2500 3250.0000
max   12.0000 6000.0000

* 1st quantile : 25% percentile is 975. that means 25% of the values less than 975.
* 2nd quantile : 50% percentile is 1600. that means 50% of the values less than 1600.
* 3rd quantile : 75% percentile is 3250. that means 75% of the values greater than 3250.